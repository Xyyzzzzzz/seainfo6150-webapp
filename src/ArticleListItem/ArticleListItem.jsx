import React from "react";

const ArticleListItem = (item) => {
    return (
        <div>
            <h1>{item.title}</h1>
            <time dateTime={item.timeStamp}>{item.displayDate}</time>
            <p>{item.shortText}</p>
        </div>
    );
}

export default ArticleListItem;