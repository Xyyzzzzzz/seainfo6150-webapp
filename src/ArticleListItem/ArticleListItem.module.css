.articleItem{
    position: relative;
    background-color: white;
    border: 1px solid brown;
    border-radius: 10px;
    margin-top: 30px;
    padding: 10px;
    box-shadow: 0px 0px 10px 10px rgb(110, 110, 110) ,
                200px 0px 100px -100px rgba(200, 0, 0, 0.5) inset;
}

.articleTitle{
    color: rgb(200, 0, 0);
    margin-top: 5px;
}

.articleTime{
    font-weight: bold;
}