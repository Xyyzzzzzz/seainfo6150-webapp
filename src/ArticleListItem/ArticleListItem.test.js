import React from "react";
import ArticleListItem from "./ArticleListItem";

const articleListItem = {
    title: "test article title",
    shortText: "test article shortText",
    timeStamp: "2018-11-22T15:12:24.000Z",
    displayDate: "November 22nd 2018, 7:12 am"
}

describe("ArticleListItem tests", () => {
    it("renders correctly", () => {
        const { container } = render(<ArticleListItem articleListItem={articleListItem} />);
        expect(container).toMatchSnapshot();
    });
});
