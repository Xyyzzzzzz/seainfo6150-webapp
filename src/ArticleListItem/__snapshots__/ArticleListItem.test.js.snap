// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <div
    class="articleItem"
  >
    <h1
      class="articleTitle"
    />
    <button
      class="toggleButton"
    >
      Show more
    </button>
  </div>
</div>
`;
